1-Draft/RFCNNNN-Download-Cmdlet.md

```diff
@@ -0,0 +1,294 @@
+---
+RFC: RFC
+Author: Mark Kraus
+Status: Draft
+SupercededBy: 
+Version: 0.1
+Area: Utility
+Comments Due: 2018-06-01
+Plan to implement: Yes
+---
+
+# Download Cmdlet
+
+This RFC is to propose the design of a new cmdlet specialized for remote file downloads to complement the existing Web Cmdlets.
+
+## Motivation
+
+    As a user of multiple protocols,
+    I can easily download remote files,
+    so that I can work with the file contents locally.
+
+The current Web Cmdlets `Invoke-WebRequest` and `Invoke-RestMethod`
+derive their value primarily from the processing they perform on the results of web requests over HTTP and HTTPS.
+This focus brings great utility to the PowerShell language by easing the methods of dealing with remote content.
+However, this focus makes it difficult to implement features that are not related to result processing.
+
+Additionally, when PowerShell Core was in development,
+there was a need to switch from the `WebRequest` API to the `HttpClient` API.
+This change greatly enhanced the flexibility and ease of coding new features for HTTP and HTTPS requests and responses.
+However, this came at the cost of support for FTP and FILE URIs.
+
+Further, as the Web Cmdlets are centered around HTTP and HTTPS,
+it would be difficult and inadvisable to implement additional protocols.
+
+There is a gap in functionality between PowerShell and other tools and shells.
+This gap was made apparent by the inclusion of `curl.exe` in Windows.
+While PowerShell and the Web Cmdlets should not seek 100% parity with similar technologies,
+The ability to download files over multiple protocols is an essential feature.
+A PowerShell user will want a PowerShell native way to perform this task.
+
+As there is a feature gap that is essential to fill
+and the current Web Cmdlets are not an ideal candidate for closing that gap,
+a new cmdlet should be added whose primary objective is to support file downloads over multiple protocols.
+
+## Specification
+
+`Invoke-Download` shall be the name of the cmdlet with the alias of `idl`.
+
+### Syntax
+
+The syntax shall be as follows:
+
+```none
+Invoke-Download -Uri <Uri> [-Credential <PSCredential>]
```

Comment:
I think it's important for this cmdlet to implement `IDynamicParameters`, and design a way to allow the type that implements `IInvokeDownloadProtocol` to provide the dynamic parameters as needed.